Assets/iCE/LightStreakFilter/Shaders/LightStreakFilter.cginc

```diff
@@ -35,6 +83,48 @@ inline float Luma(float3 color)
 	return dot(color.rgb, 0.333);
 }
 
+inline half3 UpsampleFilter(sampler2D tex, float2 uv)
+{
+    // 9-tap bilinear upsampler (tent filter)
+    float4 d = _MainTex_TexelSize.xyxy * float4(-1, -1, +1, +1);
+
+    half3 s;
+    s  = (tex2D(tex, uv + d.xy));
+    s += (tex2D(tex, uv + d.zy));
+    s += (tex2D(tex, uv + d.xw));
+    s += (tex2D(tex, uv + d.zw));
+
+    return s * (1.0 / 4);
+}
+
+inline float3 LensRefraction(sampler2D col, float2 coords)
+{
+	float3 lensRefract = float3(0.0f, 0.0f, 0.0f);
+	float2 sampleUV = float2(0.0f, 0.0f);
+
+	float2 centerTap = (coords - float2(0.5f, 0.5f));
+	centerTap *= float2((_ScreenParams.x / _ScreenParams.y) * 1.15, 1.25f) * 0.75;
+
+	for(int j = 0; j < 19; j++)
+	{
+		sampleUV = centerTap * offset[j].x;
+
+		sampleUV *= pow(2.0f * length(float2(centerTap.x, centerTap.y)), offset[j].y * 3.5f);
+		sampleUV *= offset[j].z;
+		sampleUV = float2(0.5f, 0.5f) - sampleUV;
+
+		float2 weightedRay = (sampleUV - float2(0.5f, 0.5f)) * 2.0f;
+		float lensRay = saturate(1.0f - dot(weightedRay, weightedRay));
+		float3 flare = dot(tex2D(col, sampleUV), 0.333f);
+		flare = max(0.0f, flare - 1.9f); // TODO: Make this a public variable
```

Comment:
We should resolve this TODO, and make this threshold value _separate_ from the light streak threshold value